### ISSUE TITLE
Not enough memory to run train.lua

### ISSUE THREAD

**Issue by jrriosBC** [facebookresearch/deepmask#61]
I installed deepmask with all its dependencies on a GPU instance in AWS, We picked one of the expensive ones, it was about $7/hour and it was still not able to go over the training.

we kept getting this 

"FATAL THREAD PANIC: (pcall) not enough memory"

When switching to lua 5.2 instead of luaJIT we ran into #60 and we got stuck there.

Any idea on how to work around this?

-Thanks

**Comment by tylin** [comment]
It seems like the same as https://github.com/facebookresearch/deepmask/issues/43

**Comment by caxton** [comment]
met the same issue:

th train.lua
-- ignore option rundir
-- ignore option dm
-- ignore option reload
-- ignore option gpu
-- ignore option datadir
| running in directory /home/ec2-user/deepmask/exps/deepmask/exp
| number of paramaters trunk: 15198016
| number of paramaters mask branch: 1608768
| number of paramaters score branch: 526337
| number of paramaters total: 17333121
convert: data//annotations/instances_train2014.json --> .t7 [please be patient]
convert: data//annotations/instances_train2014.json --> .t7 [please be patient]
FATAL THREAD PANIC: (pcall) not enough memory
/home/ec2-user/torch/install/bin/luajit: ...ec2-user/torch/install/share/lua/5.1/threads/threads.lua:183: [thread 2 callback] not enough memory
stack traceback:
        [C]: in function 'error'
        ...ec2-user/torch/install/share/lua/5.1/threads/threads.lua:183: in function 'dojob'
        ...ec2-user/torch/install/share/lua/5.1/threads/threads.lua:264: in function 'synchronize'
        ...ec2-user/torch/install/share/lua/5.1/threads/threads.lua:142: in function 'specific'
        ...ec2-user/torch/install/share/lua/5.1/threads/threads.lua:125: in function 'Threads'
        /home/ec2-user/deepmask/DataLoader.lua:40: in function '__init'
        /home/ec2-user/torch/install/share/lua/5.1/torch/init.lua:91: in function </home/ec2-user/torch/install/share/lua/5.1/torch/init.lua:87>
        [C]: in function 'DataLoader'
        /home/ec2-user/deepmask/DataLoader.lua:21: in function 'create'
        train.lua:101: in main chunk
        [C]: in function 'dofile'
        ...user/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
        [C]: at 0x004064c0
